lib/usd/translators/shading/usdUVTextureReader.cpp

```diff
@@ -189,8 +195,40 @@ bool PxrMayaUsdUVTexture_Reader::Read(UsdMayaPrimReaderContext* context)
             // as a relationship like texture paths inside USDZ assets.
             val = SdfAssetPath(filePath);
         }
+
+        // Re-fetch the file name in case it is UDIM-tagged
+        filePath = val.UncheckedGet<SdfAssetPath>().GetAssetPath();
         mayaAttr = depFn.findPlug(_tokens->fileTextureName.GetText(), true, &status);
         if (status == MS::kSuccess) {
+
+            // Handle UDIM dexture files:
+            std::string::size_type udimPos = filePath.rfind(_tokens->UDIMTag.GetString());
+            if (udimPos != std::string::npos) {
+                MPlug tilingAttr = depFn.findPlug(_tokens->uvTilingMode.GetText(), true, &status);
+                if (status == MS::kSuccess) {
+                    tilingAttr.setInt(3);
+
+                    // USD did not resolve the path to absolute because the file name was not an
+                    // actual file on disk. We need to find the first tile to help Maya find the
+                    // other ones.
+                    std::string udimPath(filePath.substr(0, udimPos));
+                    udimPath += "1001";
+                    udimPath += filePath.substr(udimPos + _tokens->UDIMTag.GetString().size());
+
+                    Usd_Resolver res(&prim.GetPrimIndex());
+                    for (; res.IsValid(); res.NextLayer()) {
+                        std::string resolvedName
+                            = SdfComputeAssetPathRelativeToLayer(res.GetLayer(), udimPath);
+
+                        if (!resolvedName.empty() && !ArIsPackageRelativePath(resolvedName)
+                            && resolvedName != udimPath) {
+                            udimPath = resolvedName;
+                            break;
+                        }
+                    }
+                    val = SdfAssetPath(udimPath);
```

Comment:
As of USD 20.08, there's a `UsdImaging_ResolveMaterialParamValue()` utility that might be useful here:
https://github.com/PixarAnimationStudios/USD/blob/71b4baace2044ea4400ba802e91667f9ebe342f0/pxr/usdImaging/usdImaging/materialParamUtils.h#L45

Reply:
Would love to use that function, but it is currently private:
https://github.com/PixarAnimationStudios/USD/blob/dev/pxr/usdImaging/usdImaging/CMakeLists.txt#L68
Maybe in the next version?